File: loop_sum.cpp
#include <stdlib.h>
// using namespace std; 

int main(){

int array_length = 2147483648-1;

double varsum = 0;
double x[array_length];

for (int i=0; i < array_length; i++)
{
    x[i] = rand()/(double)RAND_MAX;        
}



for (int i=0; i < array_length; i++)
{
    varsum += x[i]; 
}

return 0;
}

File: loop_sum.f90
PROGRAM looper

implicit none

integer(16) array_length, i

real(8) varsum

real(8), allocatable :: x(:)

allocate(x(2147483647-1))

varsum = 0
array_length = 2147483648-1

do i=1,array_length
    x(i) = i
end do

do i=1,array_length
        varsum = varsum + x(i)
end do

print *, varsum

deallocate(x)

END PROGRAM
